mediaire_toolbox/logging/base_logging_conf.py

```diff
@@ -48,3 +49,38 @@ def filter(self, record):
             '%(module)s:%(lineno)s %(message)s'))
 
     return logger
+
+
+def log_task_runtime(f):
+    """Function decorator for logging and recording
+    the runtime of a task in a component.
+    NOTE function with side effects; the runtime is logged
+    to the task.data dict
+    NOTE this decorator should be decorating a plain
```

Comment:
Cool idea. would it make sense to throw an Exception if the decorator is not used in the way it is meant to be used? I don't have much experience with decorators, what happens if you use it on any other function e.g. without arguments?

Reply:
Good idea. i will add some assertions.

if you use the decorator on a function that takes no arguements, then on runtime when the function is called it would throw an error saying it is taking too many positional arguements. So in our case, the decorated function should at least take in one arguement, and the passed in object should have at least 't_id', 'tag' and 'data' attributes, otherwise it would raise an error.